server/interpreter/data/object.go

```diff
@@ -16,11 +16,48 @@
 
 package data
 
-// Object represents typical JavaScript objects with (optional)
-// prototype, properties, etc.
-type Object struct {
+// Object represents any JavaScript value (primitive, object, etc.).
+type Object interface {
+	/// Any Object is a valid Value.
+	Value
+
+	// Proto returns the prototype (parent) object for this object.
+	// N.B. this is object.__proto__, not Constructor.prototype!
+	Proto() Object
+
+	// Get returns the current value of the given property or an
+	// ErrorMsg if that was not possible.
+	Get(name string) (Value, *ErrorMsg)
```

Comment:
I've gone with the first of these (since I was going to go in that direction anyway), except that I have picked the name "NativeError" rather than "Exception", to align more closely with the ECMA262 specification.  The changes are in [a subsidiary pull request](https://github.com/NeilFraser/CodeCity/pull/5) for clarity.